cloudwatch_metrics_common/test/metric_pipeline_test.cpp

```diff
@@ -184,10 +184,11 @@ MetricObject createTestMetricObject(
         const double value = 2.42,
         const std::string & unit = "gigawatts",
         const int64_t timestamp = 1234,
+        const std::map<StatisticValuesType, double> & statistic_values = std::map<StatisticValuesType, double>(),
```

Comment:
Shouldn't you create a new test rather than re-write the existing?

Reply:
I will write new test to address the decreasing code coverage.

Reply:
New test here: https://github.com/aws-robotics/cloudwatch-common/pull/48/files#diff-ea2520f48b1921373f3905fef5336ea1R212

Reply:
I don't understand how this is resolved when the test bypasses the original constructor. You would need to create two types of MetricsObjects, one with stats data and one without. Otherwise the unit tests only cover the new constructor.

Reply:
Okay, I've put back the original constructor.